controllers/configuration_controller.go

```diff
@@ -571,6 +602,48 @@ func (r *ConfigurationReconciler) preCheck(ctx context.Context, configuration *v
 		}
 	}
 
+	if meta.TerraformCredentialsSecretReference != nil {
```

Comment:
Here we have four similar logics fragment. It would be better to organize the content with a table like:
```golang
secretConfigMapToCheck:=struct{
  ref *SecretConfigMapReference
  notFoundState ConfigurationState
}{
  {gitRef, InvalidGitCredentialsSecretReference},
  {credsRef, InvalidTerraformCredentialsSecretReference
}

for _,check:=range secretConfigMapToCheck{
  // check the secret or configMap
}
```

Reply:
fixed as per review comments.